ot/torch/lp.py

```diff
@@ -79,3 +78,95 @@ def ot_solve(a, b, M, num_iter_max=100000, log=False):
         G = emd(a2, b2, M2, log=False, numItermax=num_iter_max)
 
         return torch.from_numpy(G).type_as(M)
+
+
+def ot_loss_1d(u_values, v_values, u_weights=None, v_weights=None, p=1, require_sort=True):
+    r"""
+    Computes the 1 dimensional OT loss [2] between two (batched) empirical distributions
+    ..math:
+        ot_{loss} &= \int_0^1 |cdf_u^{-1}(q)  cdf_v^{-1}(q)|^p dq
+
+    It is formally the p-Wasserstein distance raised to the power p.
+    We do so in a vectorized way by first building the individual quantile functions then integrating them.
+    This has a theoretically higher complexity than the core OT implementation but behaves better with PyTorch
+
+    Parameters
+    ----------
+    u_values: torch.Tensor (n, ...)
+        locations of the first empirical distribution
+    v_values: torch.Tensor (m, ...)
+        locations of the second empirical distribution
+    u_weights: torch.Tensor (n, ...), optional
+        weights of the first empirical distribution, if None then uniform weights are used
+    v_weights: torch.Tensor (m, ...), optional
+        weights of the second empirical distribution, if None then uniform weights are used
+    p: int, optional
+        order of the ground metric used, should be at least 1 (see [2, Chap. 2], default is 1
+    require_sort: bool, optional
+        Are the locations sorted along the last dimension already
```

Comment:
might be miss leading, True is the correct answer if the locations are _not_ sorted

Reply:
I changed it for something more explicit;

> sort the distributions atoms locations, if False we will consider they have been sorted prior to being passed to
>         the functi

on, default is True"